File: src/Deviser.Core/Deviser.Core.Common/DomainTypes/PageContent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Deviser.Core.Common.DomainTypes
{
    public class PageContent
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public Guid ContainerId { get; set; }
        public List<Property> Properties { get; set; }
        public int SortOrder { get; set; }
        public DateTime? CreatedDate { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? LastModifiedDate { get; set; }
        public Guid PageId { get; set; }
        public Guid ContentTypeId { get; set; }
        public virtual Page Page { get; set; }
        public virtual ContentType ContentType { get; set; }
        public bool InheritViewPermissions { get; set; }
        public bool InheritEditPermissions { get; set; }
        public bool HasEditPermission { get; set; }
        public virtual ICollection<PageContentTranslation> PageContentTranslation { get; set; }
        public virtual ICollection<ContentPermission> ContentPermissions { get; set; }

        public Property GetProperty(string propertyName)
        {
            if(Properties!=null && Properties.Count > 0)
            {
                return Properties.FirstOrDefault(p => p.Name == "cssclass");
            }
            return null;
        }
    }
}

File: src/Deviser.Web/ClientApp/src/app/common/domain-types/page-content.ts
import { Property } from './property';
import { ContentType } from './content-type';
import { PageContentTranslation } from './page-content-translation';
import { ContentPermission } from './content-permission';

export interface PageContent {
    id: string;
    title?: string;
    containerId: string;
    properties: Property[];
    sortOrder: number;
    createdDate?: Date | string | null;
    isActive?: boolean;
    lastModifiedDate?: Date | string | null;
    pageId: string;
    contentTypeId: string;
    // export virtual Page Page { get; set; }
    contentType?: ContentType
    inheritViewPermissions?: boolean;
    inheritEditPermissions?: boolean;
    hasEditPermission: boolean;
    pageContentTranslation?: PageContentTranslation;
    contentPermissions?: ContentPermission;
}